### PR TITLE
Sidebar updates

### PR DESCRIPTION
**Pull request by LBHLBasiu**
-sidebar sections updated 
-new content for : Dev Hub intro page, API Specification intro page, Release Notes V4
- added the  API Lifecycle Diagram 
-implement zoom in feature for images

### CODE REVIEW

**Review by humulla — APPROVED** [review]

File: docs/terraform_compliance.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+- Each API repository also holds the terraform responsible for provisioning the DynamoDB table used solely by this API to serve data.
+- The infrastructure gets provisioned during the deployment of the API.
+
+## What is Terraform Compliance?
+
+- Lightweight, security and compliance focused test framework for terraform infrastructure-as-code
+- Ensures the code implemented meets security requirements
+- Standards can be customised.
+
+![Terraform](./doc-images/terraform1.png)
```
Comment (humulla) [inline docs/terraform_compliance.md:22]: Remove Logo

File: docs/terraform_compliance.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+---
+
+## What is terraform?
+
+- Tool for building and managing infrastructure as code
+- Developers define resources that they want to provision in the cloud provider of their choice
+- Resources are defined using code in Terraform configuration files
+- Terraform generates an execution plan and is responsible for applying it (actually creating the resources) in AWS
+
+In MMH, individual services (both FE and APIs) are responsible for provisioning any infrastructure required for that specific service - for example:
```
Comment (humulla) [inline docs/terraform_compliance.md:13]: Try not to mention project names

File: docs/terraform_compliance.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+https://terraform-compliance.com/pages/bdd-references/
+![Terraform](./doc-images/terraform2.png)
+
+## How to implement it?
+
+1. Install Terraform Compliance
+2. Run Terraform Plan
+3. Run Terraform Compliance
+
+This method used is different from the other circleCI pipeline we have as normally we would run terraform init and then apply straight away. However in order to complete the terraform compliance we need to first do terraform init and plan and then run the tests to ensure the infrastructure as code is setup correctly. Once those tests passes we can run terraform apply.
```
Comment (humulla) [inline docs/terraform_compliance.md:61]: `Previously within HackIT we would first terraform init and then apply straight away. However in order to complete the terraform compliance we need to first do terraform init and plan and then run the tests to ensure the infrastructure as code is setup correctly. Once those tests pass we can run terraform apply.`

File: docs/terraform_compliance.md (hunk 1 above)
Comment (LBHLBasiu) [reply docs/terraform_compliance.md]: logo removed

File: docs/terraform_compliance.md (hunk 2 above)
Comment (LBHLBasiu) [reply docs/terraform_compliance.md]: content updated , awaiting feedback

File: docs/terraform_compliance.md (hunk 3 above)
Comment (LBHLBasiu) [reply docs/terraform_compliance.md]: content updated awaiting feedback

**Review by humulla — COMMENTED** [review]
Looks good so far :)

File: docs/lambda_best_practices.md [hunk 4]
```diff
@@ -3,32 +3,40 @@ id: lambda_best_practices
...
+Lambdas are generally best used for services that do not run for more than 5 minutes. When using Lambdas as API functions fronted by the API gateway, this time limit is further restricted to under 30 seconds (AWS imposes a hard limit on the time an API gateway endpoint can run). In the vast majority of cases this is sufficient for our API needs. There are some limitations however when we link multiple Lambdas together. Typical setup could be a front end Lambda calling a service API Lambda which in turns calls a platform API. When planning architecture around chaining different Lambdas together, it is worth analysing whether the whole process can be run well under the allocated 30 seconds.
 For the majority of cases that will be more than enough time.
 
+### When is it not best to use a Lambda function?
+
+Lambda functions should not be used for the following operations:
```
Comment (humulla) [inline docs/lambda_best_practices.md:19]: 'not' should be bold as well please

**Pull request merged**